pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/PinotConnection.java

```diff
@@ -67,15 +74,24 @@ public class PinotConnection extends AbstractBaseConnection {
     }
     _session = new org.apache.pinot.client.Connection(properties, brokers, transport);
 
-    _enableNullHandling = Boolean.parseBoolean(properties.getProperty(QueryOptionKey.ENABLE_NULL_HANDLING));
+    for (String possibleQueryOption: POSSIBLE_QUERY_OPTIONS) {
+      _queryOptions.put(possibleQueryOption, Boolean.parseBoolean(properties.getProperty(possibleQueryOption)));
+    }
   }
 
   public org.apache.pinot.client.Connection getSession() {
     return _session;
   }
 
-  public boolean isNullHandlingEnabled() {
-    return _enableNullHandling;
+  protected String enableQueryOptions(String sql) {
```

Comment:
just return query options and make a static method in `DriverUtils` to take both sql and query option map to reconstruct the query.

Reply:
+1. also is there any case we want to enable null handling but not v2. or vice-versa?

Reply:
Will do pushing the option append operation to DriverUtils. 

You can enable either or both option with this implementation. It's actually setup to allow any combination of the option keys in the Connection.POSSIBLE_QUERY_OPTIONS array.